### PR TITLE
fix: refactored measurements inner working.

### LINKED ISSUE #4145: Editing a product with additional nutrients will erroneously change the recorded values of these nutrients on save
**Describe the bug**

When editing a product with additional nutrients recorded (eg Vitamin D), the additional Nutrient values will be changed on save (even when not edited),  

**To Reproduce**
1. Edit a product with at least one additional nutrient
2. Change something (eg a Label) (but not the nutrient)
3. Save

**Expected behavior**
The nutrient value do not get changed.

**Actual behaviour**
The nutrient value is divided by 1000.

**Example**
See my edits on the product https://world.openfoodfacts.org/product/4001497280001/ : 
* Original version was https://world.openfoodfacts.org/product/4001497280001/hohes-c-orange?rev=31 − 
** `Vitamin C (ascorbic acid) | 35 mg`
* Adding a label in https://world.openfoodfacts.org/product/4001497280001/hohes-c-orange?rev=32
** `Vitamin C (ascorbic acid) | 0.035 mg`
* Adding a packaging in https://world.openfoodfacts.org/product/4001497280001/hohes-c-orange?rev=33
** `Vitamin C (ascorbic acid) | 0 mg |  `

**Smartphone (please complete the following information):**
 - Device: Fairphone 3+
 - Version 3.6.8

### PR DESCRIPTION
**Pull request by VaiTon**
Refactored inner workings for how we get/treat/send measurement (nutrition values).
Refactored code to convert measurements and to interact with them.

Use extension functions to query the db.

Closes https://github.com/openfoodfacts/openfoodfacts-androidapp/issues/4145

### Tasks
- [ ] Needs more testing (cc @teolemon)

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt [hunk 1]
```diff
@@ -69,25 +64,40 @@ open class NutrimentsGridAdapter(
...
 
-            fun fillServingValue(item: NutrimentListItem) {
-                val servingValue = item.servingValue
-                if (servingValue.isBlank()) {
+            private fun fillNutrimentValue(item: NutrimentListItem) {
+                val value = item.value
+                if (value == null) {
+                    vNutrimentValue.visibility = View.INVISIBLE
+                } else {
+                    vNutrimentValue.text = "${item.modifierStr} $value ${item.unitStr}"
```
Comment (naivekook) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt:84]: should we set visibility to visible here?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt [hunk 2]
```diff
@@ -69,25 +64,40 @@ open class NutrimentsGridAdapter(
...
+            private fun fillServingValue(item: NutrimentListItem) {
+                val servingValue = item.servingValueStr
+                if (servingValue == null) {
                     vNutrimentServingValue.visibility = View.GONE
                 } else {
-                    vNutrimentServingValue.append(String.format("%s %s %s",
-                            item.modifier,
-                            servingValue,
-                            item.unit))
+                    vNutrimentServingValue.text = "${item.modifierStr} $servingValue ${item.unitStr}"
```
Comment (naivekook) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt:94]: same question with visibility

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/additives/AdditiveFragmentHelper.kt [hunk 3]
```diff
@@ -44,14 +44,14 @@ object AdditiveFragmentHelper {
...
-                additives.forEach {
-                    append("\n")
-                    append(getAdditiveTag(it, apiClientForWikiData, fragment, fragment))
-                }
+        text = buildSpannedString {
+            bold { append(fragment.getString(R.string.txtAdditives)) }
+
+            additives.forEach {
+                append("\n")
+                append(getAdditiveTag(it, apiClientForWikiData, fragment, fragment))
```
Comment (naivekook) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/additives/AdditiveFragmentHelper.kt]: putting fragment twice here looks strange

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt (hunk 1 above)
Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt:84]: Yeah we can, despite it is already set in the XML.

**Pull request merged**